components/script_layout_interface/lib.rs

```diff
@@ -59,13 +60,27 @@ pub struct PartialPersistentLayoutData {
 
     /// Information needed during parallel traversals.
     pub parallel: DomParallelInfo,
+
+    // Required alignment in 32 bit architectures for safe
+    // transmute between PersistentLayoutData and PartialPersistentLayoutData.
+    #[cfg(target_pointer_width = "32")] _align: u64
```

Comment:
Or, better yet, make it a `[u64; 0]`, which affects alignment without size penalties.